test/Oscoin/Test/Storage/Block/Orphanage.hs

```diff
@@ -44,18 +44,12 @@ tests d =
   Properties
 ------------------------------------------------------------------------------}
 
-propInsertOrphans :: forall c. Dict (IsCrypto c) -> Property
-propInsertOrphans Dict = do
-    let orphanage    = emptyOrphanage @c blockScore
-        initialChain = unsafeToBlockchain [defaultGenesis]
-        showIt       = showOrphans . (initialChain,)
-        forkParams   = ForkParams 0 3 1  -- 1 fork of max 3 blocks.
-    forAllShow (genOrphanChainsFrom forkParams initialChain) showIt $ \chains ->
-        monadic runIdentity $ do
-            ps <- forM chains $ \(orphanChain, _missingLink) -> do
-                     let actual = toList $ getOrphans (insertOrphans (toNewestFirst $ blocks orphanChain) orphanage)
-                     pure (sort actual === sort (map blockHash (toNewestFirst $ blocks orphanChain)))
-            pure $ foldl' (.&&.) (property True) ps
+propInsertOrphansGetOrphans :: forall c. Dict (IsCrypto c) -> Property
+propInsertOrphansGetOrphans Dict =
+    forAllShow arbitraryBlockchain (toS . showChainDigest) $ \(blockchain :: Blockchain c () ())->
+        let blks = toNewestFirst $ blocks blockchain
+            orphanage = insertOrphans blks (emptyOrphanage @c blockScore)
+        in Set.fromList (map blockHash blks) === getOrphans orphanage
```

Comment:
I am actually not sure this simplification buys us anything, unfortunately 😢 When you call `arbitraryBlockchain`, you are going to create a _proper_ blockchain, i.e. a blockchain with no orphans (this might surprise you as `arbitraryBlockchain` used to do that, I think, but that generated all kinds of problems).

As a consequence of this, now you are effectively testing that you can store things inside the orphanage, but this is a bit of a moot test, as things gets interesting the minute you actually have orphans, and this is why we have the `genOrphanChainsFrom` tests 😉

Reply:
I reverted the change to the generator. We keep using `genOrphanChainsFrom`. However I still have the feeling that we don’t actually need this. I’ll think about it and we can discuss it some other time.